clang/lib/Sema/SemaDecl.cpp

```diff
@@ -6945,6 +6945,13 @@ NamedDecl *Sema::ActOnVariableDeclarator(
       NewVD->setTSCSpec(TSCS);
   }
 
+  // Static variables declared inside SYCL device code must be const or
+  // constexpr
+  if (getLangOpts().SYCLIsDevice && SCSpec == DeclSpec::SCS_static &&
+      !R.isConstant(Context))
+    SYCLDiagIfDeviceCode(D.getIdentifierLoc(), diag::err_sycl_restrict)
+        << Sema::KernelNonConstStaticDataVariable;
```

Comment:
Is this value new?  I don't remember seeing it before, but I don't see the Sema.h change to add it.

Reply:
No, It's old one.

Reply:
Then should this patch remove the old implementation of this?  Presumably this is moving a SemaSYCL check to SemaDecl, right?  If so, the old one should go away.

Reply:
No, SemaSYCL perform different check.

Reply:
Yeah, I had the same feeling. But it looks like old implementation diagnoses only usage of static data members of some class. See https://github.com/intel/llvm/blob/9a34a11cbcbf1cbb8f470bd89b3df17608fae816/clang/lib/Sema/SemaSYCL.cpp#L342 
Here, I suppose we are doing a bit other thing, but the message for `KernelNonConstStaticDataVariable` fits here very nice.

Reply:
The SemaSYCL checks static fields of class objects.

Reply:
I see.  Seems ok with me then.